Sources/NIOHTTP2/HTTP2Error.swift

```diff
@@ -17,27 +17,220 @@ public protocol NIOHTTP2Error: Equatable, Error { }
+    }
+
+    public static func missingHostHeader(file: String = #file, line: UInt = #line) -> MissingHostHeader {
+        return MissingHostHeader(file: file, line: line)
+    }
+
+    public static func duplicateHostHeader(file: String = #file, line: UInt = #line) -> DuplicateHostHeader {
+        return DuplicateHostHeader(file: file, line: line)
+    }
+
+    public static func emptyPathHeader(file: String = #file, line: UInt = #line) -> EmptyPathHeader {
+        return EmptyPathHeader(file: file, line: line)
+    }
+
+    public static func invalidStatusValue(_ value: String, file: String = #file, line: UInt = #line) -> InvalidStatusValue {
+        return InvalidStatusValue(value, file: file, line: line)
+    }
+
+    public static func priorityCycle(streamID: HTTP2StreamID, file: String = #file, line: UInt = #line) -> PriorityCycle {
+        return PriorityCycle(streamID: streamID, file: file, line: line)
+    }
+
+    public static func trailersWithoutEndStream(streamID: HTTP2StreamID, file: String = #file, line: UInt = #line) -> TrailersWithoutEndStream {
+        return TrailersWithoutEndStream(streamID: streamID, file: file, line: line)
+    }
+
+    public static func invalidHTTP2HeaderFieldName(_ fieldName: String, file: String = #file, line: UInt = #line) -> InvalidHTTP2HeaderFieldName {
+        return InvalidHTTP2HeaderFieldName(fieldName, file: file, line: line)
+    }
+
+    public static func forbiddenHeaderField(name: String, value: String, file: String = #file, line: UInt = #line) -> ForbiddenHeaderField {
+        return ForbiddenHeaderField(name: name, value: value, file: file, line: line)
+    }
+
+    public static func contentLengthViolated(file: String = #file, line: UInt = #line) -> ContentLengthViolated {
+        return ContentLengthViolated(file: file, line: line)
+    }
+
+    public static func excessiveEmptyDataFrames(file: String = #file, line: UInt = #line) -> ExcessiveEmptyDataFrames {
+        return ExcessiveEmptyDataFrames(file: file, line: line)
+    }
+
+    public static func excessivelyLargeHeaderBlock(file: String = #file, line: UInt = #line) -> ExcessivelyLargeHeaderBlock {
+        return ExcessivelyLargeHeaderBlock(file: file, line: line)
+    }
+
+    public static func noStreamIDAvailable(file: String = #file, line: UInt = #line) -> NoStreamIDAvailable {
+        return NoStreamIDAvailable(file: file, line: line)
+    }
+
     /// The outbound frame buffers have become filled, and it is not possible to buffer
     /// further outbound frames. This occurs when the remote peer is generating work
     /// faster than they are consuming the result. Additional buffering runs the risk of
     /// memory exhaustion.
     public struct ExcessiveOutboundFrameBuffering: NIOHTTP2Error {
-        public init() { }
+        /// The file and line where the error was created.
+        public var file: String
+
+        @available(*, deprecated, renamed: "excessiveOutboundFrameBuffering")
```

Comment:
I think these need the fully-qualified function name, as this function is not on the same type as its replacement and so the fixit won't work.

Reply:
I thought that'd be the case as well, but Xcode turns instances of `NIOHTTP2Errors.ExcessiveOutboundFrameBuffering` into `NIOHTTP2Errors.NIOHTTP2Errors.excessiveOutboundFrameBuffering`!

FWIW all the replacements in this PR were done by applying fixits.

Reply:
Weird, ok then!